app/controllers/panel_controller.rb

```diff
@@ -50,6 +50,48 @@ def generate_db_token
     }
   end
 
+  private def validators_for_competition_ids(competition_ids)
+    validators = params.require(:selectedValidators).split(',').map(&:constantize)
+    apply_fix_when_possible = params.require(:applyFixWhenPossible)
+
+    results_validator = ResultsValidators::CompetitionsResultsValidator.new(
+      validators,
+      check_real_results: true,
+      apply_fixes: apply_fix_when_possible,
+    )
+    results_validator.validate(competition_ids)
+    render json: {
+      has_results: results_validator.has_results?,
+      validators: results_validator.validators,
+      infos: results_validator.infos,
+      errors: results_validator.errors,
+      warnings: results_validator.warnings,
+    }
+  end
+
+  private def competition_ids_in_range(range)
+    start_date = range[:startDate]
+    end_date = range[:endDate]
+    Competition.over
+               .not_cancelled
+               .where(start_date: start_date..)
+               .where(start_date: ..end_date)
+               .order(:start_date)
+               .ids
```

Comment:
Can you combine this with the logic for `competition_count` above?
Like, have some `competitions_between(start, end)` method which returns `Competition.over.not_cancelled.where(...).order(...)` and then one endpoint returns `competitions_between(foo, bar).count` and the other one returns `competitions_between(foo, bar).ids`

Reply:
This seems like the perfect use case for an AR `scope` by the way :)